projects/js-packages/components/components/alert/index.tsx

```diff
@@ -19,9 +18,20 @@ export const LEVEL_WARNING = 'warning';
 export const LEVEL_INFO = 'info';
 export const LEVEL_SUCCESS = 'success';
 
-export const ALERT_LEVELS = [ LEVEL_ERROR, LEVEL_WARNING, LEVEL_INFO, LEVEL_SUCCESS ];
+export const ALERT_LEVELS = [ LEVEL_ERROR, LEVEL_WARNING, LEVEL_INFO, LEVEL_SUCCESS ] as const;
```

Comment:
Just a question that forgot to do, this `ALERT_LEVELS` is going to be used by some consumers? Or just at storybook? 

Saying that because if is going to be used by consumers, an array is probably not the best at dev ex, and if is only for storybook, you could define it only there, right?

Reply:
> ..., an array is probably not the best at dev ex

It's Debatable? :-), and depends on the implementation. Having the Array and also constants for all items is enough to avoid the primary issue, which is a misspelling or a wrong level name.

> ... and if is only for storybook
Yes at least so far. But considering that the component can be used (in fact it does) by other components no-written-in-ts, probably we'd like to expose them anyway? I consider it a transition step.

Reply:
> It's Debatable?

Sure. It is. I say it is not the best dev ex because to use it you need to do: `ALERT_LEVELS[0]`, which makes it strange, since I need to know each one index, instead of doing `AlertLevels.ERROR` for example. And If I already export each one constant, it makes the array unneeded, not?

>  But considering that the component can be used (in fact it does) by other components no-written-in-ts, probably we'd like to expose them anyway?

Don't get this point, TS already provides interoperability with JS, so we don't need to care about that kind of thing.

Reply:
> If I already export each one constant, it makes the array unneeded, not?

I agree with @renatoagds here.

Storybook reflects the actual usage by a consumer. So we don't expect the consumer components to use that array.

I would still suggest not to create any of those constants at all and rather use Union of string literals to reduce the runtime cost of this component. These small bits can hugely affect the bundle size if this is made a widespread pattern accross different components.

Reply:
> ...to use it you need to do: `ALERT_LEVELS[0]`,

I don't expect that usage.

> And If I already export each one constant, it makes the array unneeded, not?

Sometimes it could be useful. For instance, in the storybook.

> ...so we don't need to care about that kind of thing.

I can define a invalid prop value and it won't be detected:

<img width="378" alt="image" src="https://user-images.githubusercontent.com/77539/166709863-562ff235-70b8-400f-a529-6b5110a93727.png">

No error below because the IDE it isn't a ts file. However, in ts files:

<img width="466" alt="image" src="https://user-images.githubusercontent.com/77539/166710638-80703751-ace8-47c2-a69f-c265a4d070fd.png">

Reply:
addressed here https://github.com/Automattic/jetpack/pull/24204/commits/96f43442349b1d277a110912245e71b6defeb0f8